### PR TITLE
Replace LRR module with generic RSS module

### LINKED ISSUE #36: Refactor LRR module into a generic RSS feed checker
And move all the feeds into the database (#13), when that is implemented.

Had the idea in #14.

### PR DESCRIPTION
**Pull request by HubbeKing**
Closes #36, assuming it works? Should. The admin-checking is maybe a bit wonk?

### CODE REVIEW

File: desertbot/modules/commands/RSS.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                               message.replyTo)
+        else:
+            return IRCResponse(ResponseType.Say,
+                               "I am not following any feed named {}".format(name),
+                               message.replyTo)
+
+    def _listFeeds(self, message: IRCMessage):
+        return IRCResponse(ResponseType.Say,
+                           "Currently followed feeds are: {}".format(", ".join(self.feeds.keys())),
+                           message.replyTo)
```
Comment (StarlitGhost) [inline desertbot/modules/commands/RSS.py:205]: you're missing the rss = RSS() at the bottom :P

**Review by StarlitGhost — CHANGES_REQUESTED** [review]

File: desertbot/modules/commands/RSS.py (hunk 1 above)
Comment (HubbeKing) [reply desertbot/modules/commands/RSS.py:205]: oh I am a smart

File: desertbot/modules/commands/RSS.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+            name = " ".join(message.parameterList[2:])
+            feed_object = {
+                "url": url,
+                "lastUpdate": datetime.datetime.utcnow() - datetime.timedelta(days=365),
+                "lastTitle": "",
+                "lastLink": "",
+                "lastCheck": datetime.datetime.utcnow() - datetime.timedelta(minutes=10),
+                "suppress": False
+            }
+            self.feeds.append({name: feed_object})
```
Comment (StarlitGhost) [inline desertbot/modules/commands/RSS.py]: dicts don't have an append method, I guess you just want self.feeds[name] = feed_object

**Comment by HubbeKing** [comment]
Travis build failure is unrelated, see #146

File: desertbot/modules/commands/RSS.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                return self._unfollowFeed(message)
+            elif message.parameterList[0].lower() == "toggle":
+                return self._toggleFeedSuppress(message)
+            elif message.parameterList[0].lower() == "list":
+                return self._listFeeds(message)
+            elif len(message.parameters.strip()) > 0:
+                feed = message.parameters.strip()
+                latest = self._getLatest(feed)
+                if latest is not None:
+                    response = 'Latest {}: {} | {}'.format(feed["name"], feed["title"], feed["link"])
```
Comment (StarlitGhost) [inline desertbot/modules/commands/RSS.py]: `latest`, not `feed`

File: desertbot/modules/commands/RSS.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+                continue
+
+            itemDate = item.find("pubdate").text
+            newestDate = dparser.parse(itemDate, fuzzy=True, ignoretz=True).isoformat()
+
+            if newestDate > feedDeets["lastUpdate"]:
+                self.feeds[feedName]["lastUpdate"] = newestDate
+                title = item.find("title").text
+                link = item.find("link").text
+                link = self.bot.moduleHandler.runActionUntilValue("shorten-url", link)
```
Comment (Heufneutje) [inline desertbot/modules/commands/RSS.py]: This should probably fall back on the long link if URL shortening fails for whatever reason.

**Review by Heufneutje — CHANGES_REQUESTED** [review]
Just one little thing. Looks great otherwise.

**Comment by HubbeKing** [comment]
Should be fixed and working now, seemed to work on the testnet at least.

**Comment by HubbeKing** [comment]
The only things remaining is that calling `.rss <feed_name>` doesn't update the feed if there's been no other messages in a channel recently, since updating is triggered by `message-channel`. This means in order to ACTUALLY get latest from the RSS feed, it'd need to be >10 minutes since the last update, and you'd need to send a non-rss command or other message first, and then do `.rss <feed_name>`

It would be handy to update feeds on something else, so there'd be updates even if channels are quiet, but then you'd need a way to specify where to send notifications of new posts.

**Comment by HubbeKing** [comment]
Okay, the module now triggers an update if needed when queried for latest item for a feed, and automatically fetches the latest item when a new feed is added.
Probably good enough? It'd be neat to update on things that aren't `message-channel`, but that's a bit more work.

**Review by StarlitGhost — APPROVED** [review]

**Pull request merged**